index.bs

```diff
@@ -12366,14 +12366,21 @@ object is associated with.
 <div algorithm>
   To <dfn export lt="new|create|create a new|create an object implementing|create an object
   implementing the interface">create an object implementing the interface</dfn> |interface| in the
-  Realm |realm|, perform the following steps:
+  Realm |realm|, with optional `new.target` value |newTarget|, perform the following steps:
 
     1.  Assert: |interface| is [=exposed=] in |realm|.
+    1.  Let prototype be the [=interface prototype object=] for |interface| in
+        |realm|.
+    1.  If |newTarget| is provided, then:
+        1.  Issue: If we just pass {{NewTarget}} through in a constructor, would this not
+            always be defined (sometimes to the [=interface object=] itself)?
+        1.  Let |newTargetPrototype| be [=?=] [$Get$](|newTarget|, "prototype").
```

Comment:
The logic here is a bit different from the JS spec, which is based on [OrdinaryCreateFromConstructor](https://tc39.github.io/ecma262/#sec-ordinarycreatefromconstructor).

Actually, it seems like we've opened up here a whole world of divergence between JS and the Web, as well as between browsers. Consider this test case:

```js
function foo() { }
function bar() { } bar.prototype = 1;

console.log(`
  ${ Reflect.construct(URL, ["file:///foo"], foo).constructor.name }
  ${ Reflect.construct(URL, ["file:///foo"], bar).constructor.name }
  ${ Reflect.construct(Uint8Array, [1], foo).constructor.name }
  ${ Reflect.construct(Uint8Array, [1], bar).constructor.name }
`);
```

WebKit results:

```
  URL
  URL
  foo
  Uint8Array
```

Chromium results:

```
  foo
  Object
  foo
  Uint8Array
```

Gecko results:

```
  foo
  URL
  foo
  Uint8Array
```

The Gecko results have the closest analogy to JavaScript here--they are as if we're following the OrdinaryCreateFromConstructor algorithm, with all of the platform interfaces included in the fallback list. If Chromium and WebKit folks are in favor of it, I'd suggest adopting these semantics.

Reply:
Given the three-way divergence, and the extreme edge-casey nature of this `Reflect.construct` madness, I think it'll be pretty safe to pick one, write tests, and file bugs on everyone else. So I encourage us to pick Gecko. /cc @yuki3 @cdumez in case they want to weigh in, but IMO we shouldn't block.

Note that for custom elements we tried to copy ES more exactly: https://html.spec.whatwg.org/multipage/dom.html#html-element-constructors steps 7 and 8.

Reply:
No objection.  Please go ahead.  :)

Reply:
Yeah, to be clear, the Gecko behavior, in the absence of an explicit spec, was written to match the ES semantics, since that was the only example of constructor behavior around.  I don't see any reason for deviating from those semantics in Web IDL, but obviously I'm biased, since I wrote the Gecko code here. ;)